apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php

```diff
@@ -434,9 +435,28 @@ public function get_close_button_url() {
 		 *
 		 * @param string Current close button URL.
 		 */
-		$close_button_url = apply_filters( 'a8c_fse_close_button_link', $close_button_url );
+		return apply_filters( 'a8c_fse_close_button_link', $close_button_url );
+	}
+
+	/**
+	 * Returns the base URL for the Edit Template Part button. The URL does not contain neither
+	 * the post ID nor the template part ID. Those query arguments should be provided by
+	 * the Template Part on the Block.
+	 *
+	 * @return string edit link without post ID
+	 */
+	public function get_edit_template_part_base_url() {
+		$edit_post_link = esc_url( remove_query_arg( 'post', get_edit_post_link( 0, 'edit' ) ) );
 
-		return $close_button_url;
+		/**
+		 * Filter the Gutenberg's edit template part button base URL
+		 * when editing pages or posts.
+		 *
+		 * @since 0.2
+		 *
+		 * @param string Current edit button URL.
+		 */
+		return apply_filters( 'a8c_fse_edit_template_part_base_url', $edit_post_link );
```

Comment:
We should try to [escape as late as possible](https://codex.wordpress.org/Validating_Sanitizing_and_Escaping_User_Data); this way, anything coming from the filter is caught here too.

```suggestion
		return esc_url( apply_filters( 'a8c_fse_edit_template_part_base_url', $edit_post_link ) );
```

Reply:
haven't seen this in the wild, so I've added `esc_url` when we use `wp_localize_script` instead.

https://github.com/Automattic/wp-calypso/blob/68a424d62fd008997b93f6823ed6cb69c21eb680/apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php#L307-L312

updated on 68a424d62fd008997b93f6823ed6cb69c21eb680.